CMakeLists.txt

```diff
@@ -102,10 +111,68 @@ find_package(Boost COMPONENTS filesystem)
 include_directories(${Boost_INCLUDE_DIRS})
 
 # TODO: utils-urdf should only be required for aikido::perception.
-find_package(DART REQUIRED COMPONENTS optimizer-nlopt utils-urdf CONFIG)
+find_package(DART 6.1.0 REQUIRED COMPONENTS optimizer-nlopt utils-urdf CONFIG)
 include_directories(${DART_INCLUDE_DIRS})
 
 #==============================================================================
+# Compiler settings
+#
+if(CMAKE_COMPILER_IS_GNUCXX)
+
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8)
+    message(FATAL_ERROR "The installed g++ version is
+        ${CMAKE_CXX_COMPILER_VERSION}. ${PROJECT_NAME} requires g++ 4.8 or
+        greater."
+    )
+  endif()
+
+  if(TREAT_WARNINGS_AS_ERRORS)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+  endif()
+
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.3)
+    message(FATAL_ERROR "The installed Clang version is
+        ${CMAKE_CXX_COMPILER_VERSION}. ${PROJECT_NAME} requires clang 3.3 or
+        greater."
+    )
+  endif()
+
+  if(TREAT_WARNINGS_AS_ERRORS)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+  endif()
+
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.1)
+    message(FATAL_ERROR "The installed AppleClang version is
+        ${CMAKE_CXX_COMPILER_VERSION}. ${PROJECT_NAME} requires AppleClang 6.1
+        or greater."
+    )
+  endif()
+
+  if(TREAT_WARNINGS_AS_ERRORS)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+  endif()
+
+elseif(MSVC)
+
+  if(MSVC_VERSION VERSION_LESS 1900)
+     message(FATAL_ERROR "${PROJECT_NAME} requires VS 2015 or greater.")
+   endif()
+
+  if(TREAT_WARNINGS_AS_ERRORS)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /WX")
+  endif()
+
+else()
+
+  message(SEND_ERROR "Compiler[${CMAKE_CXX_COMPILER_ID}] not supported.")
```

Comment:
I actually prefer to have all the compiler settings in a helper file like `CompilerSettings.cmake` as it will be a big block of script. What do you think?